### PR TITLE
Convert contribution owner to ID, use smaller number formats

### PR DESCRIPTION
**Pull request by skddc**
Sorry for mixing these two topics into one PR. This changes the contribution to be tied to a contributor ID instead of an account address. It also changes a bunch of numbers from uint256 to uint32.

### CODE REVIEW

File: apps/contribution/contracts/Contribution.sol [hunk 1]
```diff
@@ -4,21 +4,29 @@ import "@aragon/os/contracts/apps/AragonApp.sol";
...
+
   // ensure alphabetic order
   enum Apps { Contribution, Contributor, Proposal, Token }
   bytes32[4] public appIds;
 
   struct ContributionData {
-    address contributor;
-    uint256 amount;
+    uint32 contributorId;
+    uint32 amount;
```
Comment (fsmanuel) [inline apps/contribution/contracts/Contribution.sol]: The `mintFor` takes a `uint256`

File: lib/contracts/contribution.js [hunk 2]
```diff
@@ -20,24 +19,40 @@ class Contribution extends Base {
...
-      .then((data) => {
+      .then(data => {
         return this.ipfs.catAndMerge(data, ContributionSerializer.deserialize);
       });
 
   }
 
-  getByContributor(contributor) {
-    return this.functions.balanceOf(contributor)
+  getByContributorId(contributorId) {
```
Comment (fsmanuel) [inline lib/contracts/contribution.js]:
```suggestion
getByContributorId(contributorId) {
  return this.functions.getContributorAddressById(contributorId)
    .then(address => this.getByContributorAddress(address));
}
```

File: lib/contracts/contribution.js [hunk 3]
```diff
@@ -20,24 +19,40 @@ class Contribution extends Base {
...
+    //         });
+    //     }
+    //
+    //     return RSVP.all(contributions);
+    //   });
+  }
+
+  getByContributorAddress(address) {
+    return this.functions.balanceOf(address)
       then((balance) => {
```
Comment (fsmanuel) [inline lib/contracts/contribution.js]: It is missing a `.`

File: apps/contribution/contracts/Contribution.sol (hunk 1 above)
Comment (skddc) [reply apps/contribution/contracts/Contribution.sol]: Yes, but we're converting it before minting in `claim()`. Nobody has to mine more than 2.2 billion kredits at once, so the locally stored amount can be a smaller format.

File: lib/contracts/contribution.js (hunk 3 above)
Comment (skddc) [reply lib/contracts/contribution.js]: I may be blind, but where exactly? Maybe you could turn it into a suggestion.

File: lib/contracts/contribution.js [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
 class Contribution extends Base {
   all() {
     return this.functions.contributionsCount()
-      .then(count => {
+      .then(async (count) => {
         let contributions = [];
 
         for (let id = 1; id <= count; id++) {
-          contributions.push(this.getById(id));
+          const contribution = await this.getById(id)
```
Comment (fsmanuel) [inline lib/contracts/contribution.js]: ❤️

Comment (bumi) [reply lib/contracts/contribution.js]: to promise or not to promise? :bumiwonders:

Comment (fsmanuel) [reply lib/contracts/contribution.js]: If you can wait for it you don't need to promise anything.

**Comment by skddc** [comment]
fuck-yeah

**Pull request merged**